frappe/public/js/frappe/utils/utils.js

```diff
@@ -1783,13 +1783,11 @@ Object.assign(frappe.utils, {
 	 * @returns {boolean} - Returns `true` if the value is empty, `false` otherwise.
 	 */
 	is_empty(value) {
-		if (value && typeof value === "object") {
-			return Object.keys(value).length === 0;
-		} else if (value === 0) {
-			// value of an unchecked checkbox
-			return false;
-		} else {
-			return !value;
-		}
+		if (!value && value !== 0) return true;
+
+		if (typeof value === "object")
+			return (Array.isArray(value) ? value : Object.keys(value)).length === 0;
```

Comment:
@sagarvora why this more complicated statement? Is there a case where `my_array.length !== Object.keys(my_array).length`?

Reply:
Computing `Object.keys` is unnecessary for an Array. It just returns a new Array of indices of the existing Array.

Array itself has length.